users/email/email.go

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package email
+
+import (
+	"fmt"
+	"log"
+	"net/smtp"
+	"os"
+	"sync"
+
+	"github.com/mainflux/mainflux"
+	"github.com/mainflux/mainflux/logger"
+)
+
+const (
+	defEmailLogLevel    = "debug"
+	defEmailDriver      = "smtp"
+	defEmailHost        = "localhost"
+	defEmailPort        = "25"
+	defEmailUsername    = "root"
+	defEmailPassword    = ""
+	defEmailFromAddress = ""
+	defEmailFromName    = ""
+
+	envEmailDriver      = "MF_EMAIL_DRIVER"
+	envEmailHost        = "MF_EMAIL_HOST"
+	envEmailPort        = "MF_EMAIL_PORT"
+	envEmailUsername    = "MF_EMAIL_USERNAME"
+	envEmailPassword    = "MF_EMAIL_PASSWORD"
+	envEmailFromAddress = "MF_EMAIL_FROM_ADDRESS"
+	envEmailFromName    = "MF_EMAIL_FROM_NAME"
+	envEmailLogLevel    = "MF_EMAIL_LOG_LEVEL"
+)
+
+type email struct {
+	driver      string
+	host        string
+	port        string
+	username    string
+	password    string
+	fromAddress string
+	fromName    string
+}
+
+// Agent for mailing
+type Agent struct {
+	conf email
+	auth smtp.Auth
+	addr string
+	log  logger.Logger
+}
+
+var a *Agent
+var once sync.Once
+
+// Instance - Thread safe creation of mail agent
+func Instance() *Agent {
+	once.Do(func() {
```

Comment:
I'm just curious, why do you use `once.Do`?

Reply:
to ensure singleton